cloudinit/config/cc_snap.py

```diff
@@ -110,6 +118,12 @@
                     'additionalItems': False,  # Reject non-string & non-list
                     'minItems': 1,
                     'minProperties': 1,
+                    'additionalProperties': {
+                        'oneOf': [
```

Comment:
This works for a lot of cases, but still doesn't seem to reject invalid cloud-config

```
cat > invalid-snap-item-0.yaml <<EOF
#cloud-config
snap:
 commands:
   0: -1
   0: ['install', 'vlc']
   1: snap install vlc
   2: "snap install vlc"
EOF
python3 -m cloudinit.cmd.main devel schema -c invalid-snap-item-0.yaml
Valid cloud-config file invalid-snap-item-0.yaml